### PR TITLE
Optimization fix + use swig tag

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

File: lib/src/Base/Optim/Cobyla.cxx [hunk 1]
```diff
@@ -123,7 +123,7 @@ void Cobyla::run()
    */
   int returnCode = ot_cobyla(n, m, &x[0], rhoBeg_, rhoEnd, message, &maxFun, Cobyla::ComputeObjectiveAndConstraint, (void*) this);
 
-  result_ = OptimizationResult(dimension, 2);
+  result_ = OptimizationResult(dimension, 1);
```
Comment (dbarbier) [inline lib/src/Base/Optim/Cobyla.cxx]: Looks wrong to me, see l.101
```
  // initialize history
  evaluationInputHistory_ = Sample(0, dimension);
  evaluationOutputHistory_ = Sample(0, 2);
```

Comment (dbarbier) [reply lib/src/Base/Optim/Cobyla.cxx]: Okay, got it now

Comment (jschueller) [reply lib/src/Base/Optim/Cobyla.cxx]: in evaluationOutputHistory_ we store the value + constraint so its of dim 2 yes

my modification changes the dimension of the outputHistory_ of OptimResult:
```result_.store(inP, Point(1, outP[0])```

**Comment by jschueller** [comment]
@regislebrun could you review this one ? it moves away from swig master and would fix our circleci builds.

**Comment by regislebrun** [comment]
+1

**Pull request merged**